conans/model/settings.py

```diff
@@ -355,7 +362,10 @@ def constraint(self, constraint_def):
         # Sanity check for input constraint wrong fields
         for field in constraint_def:
             if field not in self._data:
-                raise undefined_field(self._name, field, self.fields)
+                if raise_missing_value:
+                    raise undefined_field(self._name, field, self.fields)
+                else:
+                    fields_to_remove.append(field)
```

Comment:
This can fail. You cannot remove fields that are not defined in current settings.

Reply:
ok, will add a test

Reply:
The removal is already protected, won't fail, but still ugly.

Probably the solution is just:
```python
if raise_missing_value:
  for field in constraint_def:
    if field not in self._data:		           
       raise undefined_field(self._name, field, self.fields)		
```

I am thinking about other implications